docs/src/proposals/scrambled-tx.md

```diff
@@ -0,0 +1,335 @@
+
+#### Validators
+
+They want cluster fork less frequently so generally cooperates.
+Unless significant nodes (like more than 1/3) are colluding, they pre-vote
+blocks honestly with accurate wall clock in the fear of reduced rewards because
+of being outstanding outlier otherwise.
+
+Also they're incentivised for publishing scrambled transactions as first solver
+by small rewards. Lying the results is eligible for slashing to ensure bad
+scrambled transaction is detected and backed by stakes.
+
+### Further extensions
+
+This proposal lays down the foundation for further protocol extension.
+
+Firstly, this whole new descrambling process can be regarded as a deterministic
+and async commit-and-reveal. This means a non-interactive, unbiasble,
+deterministic and free-as-in-beer random number can be provided at the
+consensus layer as by-product, with a few tweaks of this proposal.
+
+That random can easily be fed into transaction runtime for one thing. Further,
+this random number is also used to combat against verifiers's dilemma (or
+tragedy of commons) as well, because of the very existence at the consensus
+layer, opening fierce yet cooperative validator competition.
+
+Assuming all relevant parties are directed towards their each own profit
+maximization according to incentives in this proposal, MEV players will compete
+fully on chain as the new and last battle frontier for any conceivable edges,
+using the on-chain transaction submission mechanism for smallest latency, which
+will be introduced in the future. This will nullify co-location advantage with
+highly-staked validators by private deals (or the first-hand MEV-ing by the
+validators, exploiting the block-production privilege).
+
+All in all, the induced latency should be worth as the trade-off to realize
+these technical breakthroughs.
+
+### Alternative approch
+
+#### multiple block producers concurrently.
+
+todo: and collect and analyze anatoly's thoughts.
+
+also, this proposal is trying even to work for realizaton of fair palys for on-chain quiz.
+
+stake centralization? ohoh, scrambled block incentivise just high uptime for validator to earn more.
+
+#### possible off-chain marketplace like flashbots
+
+should be economically inferior because of existense of this unified on-chain one.
+
+### Prior work
+
+None? This scrambling is leveraging the fact Solana's very short block time and
+abundance of online GPU resources.
+
+### Remaining concerns
+
+- Unsolved MEV opportunity on the on-chain mempool. So clients should be wise to
+  value their transaction with adequate additional priority fee.
+
+- Reveal of intension from failed scrambled transaction. For that concern,
+  strong (= finger-printing resistant) scrambling is needed. also, this can tip
+  about more-or-less immediate general market sentiment change after some news
+  release, even if these transactions are individually protected from
+  reordering?
+
+- sha256-ing throughput difference between cpu and gpu?
```

Comment:
on the other hand, this scrambling should reinforce asic resistance of poh as a whole?